hax/hax/motr/__init__.py

```diff
@@ -341,11 +343,24 @@ def ha_obj_state(st):
                                                             kv_cache=kv_cache)
         if not notes:
             return []
-        message_ids: List[MessageId] = self._ffi.ha_broadcast(
-            self._ha_ctx, make_array(HaNoteStruct, notes), len(notes))
-        LOG.debug(
-            'Broadcast HA state complete with the following message_ids = %s',
-            message_ids)
+        notes_pending = notes
+        notes_len_total = len(notes)
+        idx = 0
+        message_ids: List[MessageId] = []
+        while notes_pending:
+            if len(notes_pending) > MAX_MOTR_NVEC_UPDATE_SZ:
+                notes_to_send = notes[idx:(MAX_MOTR_NVEC_UPDATE_SZ - 1)]
```

Comment:
The second argument is not the length of the resulting slice rather than `stop` index. [Quote from SO](https://stackoverflow.com/questions/509211/understanding-slice-notation):
```
a[start:stop]  # items start through stop-1
a[start:]      # items start through the rest of the array
a[:stop]       # items from the beginning through stop-1
a[:]           # a copy of the whole array
```

Which makes me think that this algorithm will work for lengths <= 2048.

Suggestion:
```python
while notes:
    notes_to_send = notes[:MAX_MOTR_NVEC_UPDATE_SZ] # take first MAX_MOTR_NVEC_UPDATE_SZ - 1 elements
    <send notes_to_send>
    notes = notes[MAX_MOTR_NVEC_UPDATE_SZ:] # take the rest of notes - cut the first MAX_MOTR_NVEC_UPDATE_SZ-1 elements

Reply:
Updated the patch also added an unit test. Dry run of the algorithm,
```>>> x_to_send
[1, 2, 3, 4]
>>> idx += len(x_to_send)
>>> x_to_send = x[idx:(idx + max_sz)]
>>> x_to_send
[5, 6, 7, 8]
>>> idx += len(x_to_send)
>>> x_to_send = x[idx:(idx + max_sz)]
>>> x_to_send
[9, 10]
>>> idx += len(x_to_send)
>>> x_to_send = x[idx:(idx + max_sz)]
>>> x_to_send
[]
>>>
```